semanticdb/scalac/library/src/main/scala/scala/meta/internal/semanticdb/scalac/TextDocumentOps.scala

```diff
@@ -595,16 +637,16 @@ trait TextDocumentOps { self: SemanticdbOps =>
       val finalSymbols = symbols.values.toList
 
       val finalOccurrences = {
-        occurrences.flatMap {
-          case (pos, sym) =>
-            sym.asMulti.map { flatSym =>
-              val role =
-                if (binders.contains(pos)) s.SymbolOccurrence.Role.DEFINITION
-                else s.SymbolOccurrence.Role.REFERENCE
-              s.SymbolOccurrence(Some(pos.toRange), flatSym, role)
-            }
-        }.toList
-      }
+        for {
+          (pos, sym) <- Iterator(occurrences, mpatoccurrences).flatten
+          flatSym <- sym.asMulti
+        } yield {
+          val role =
+            if (binders.contains(pos)) s.SymbolOccurrence.Role.DEFINITION
+            else s.SymbolOccurrence.Role.REFERENCE
+          s.SymbolOccurrence(Some(pos.toRange), flatSym, role)
+        }
+      }.toList
```

Comment:
This reads like black magic, but that's because the entire TextDocumentOps is black magic. I'm extremely happy that we have a great test suite, because otherwise I don't see how we could be 100% sure that these changes do the right thing in all cases. Special thanks for finding the right incantation to fix the bug without breaking everything else.